pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -265,22 +266,35 @@ func ValidateNodeAffinityArgs(args *config.NodeAffinityArgs) error {
 		return nil
 	}
 	affinity := args.AddedAffinity
-	f := field.NewPath("addedAffinity")
+	path := field.NewPath("addedAffinity")
 	var allErrs field.ErrorList
 	if ns := affinity.RequiredDuringSchedulingIgnoredDuringExecution; ns != nil {
-		_, err := nodeaffinity.NewNodeSelector(ns)
+		_, err := nodeaffinity.NewNodeSelector(ns, nodeaffinity.WithPath(path.Child("requiredDuringSchedulingIgnoredDuringExecution")))
 		if err != nil {
-			// TODO(#96167): Expand all field.Error(s) returned once constructor use them.
-			allErrs = append(allErrs, field.Invalid(f.Child("requiredDuringSchedulingIgnoredDuringExecution"), affinity.RequiredDuringSchedulingIgnoredDuringExecution, err.Error()))
+			allErrs = appendFieldErrorListFromErr(allErrs, err)
 		}
 	}
 	// TODO: Add validation for requiredDuringSchedulingRequiredDuringExecution when it gets added to the API.
 	if terms := affinity.PreferredDuringSchedulingIgnoredDuringExecution; len(terms) != 0 {
-		_, err := nodeaffinity.NewPreferredSchedulingTerms(terms)
+		_, err := nodeaffinity.NewPreferredSchedulingTerms(terms, nodeaffinity.WithPath(path.Child("preferredDuringSchedulingIgnoredDuringExecution")))
 		if err != nil {
-			// TODO(#96167): Expand all field.Error(s) returned once constructor use them.
-			allErrs = append(allErrs, field.Invalid(f.Child("preferredDuringSchedulingIgnoredDuringExecution"), affinity.PreferredDuringSchedulingIgnoredDuringExecution, err.Error()))
+			allErrs = appendFieldErrorListFromErr(allErrs, err)
 		}
 	}
 	return allErrs.ToAggregate()
 }
+
+func appendFieldErrorListFromErr(errs field.ErrorList, err error) field.ErrorList {
+	agg, ok := err.(errors.Aggregate)
+	if !ok {
+		panic("err is not an errors.Aggregate")
+	}
+	for _, e := range agg.Errors() {
+		fErr, ok := e.(*field.Error)
+		if !ok {
+			panic("wrapped err is not a field.Err")
+		}
+		errs = append(errs, fErr)
```

Comment:
seems like we'd still be ok just appending errors we don't recognize as special types... is just flattening the aggregated error using `errors.Flatten` problematic?

Reply:
You are right, we shouldn't care about the type. I didn't know about `errors.Flatten`. It's very handy.